plugins/hydra_ax_sweeper/tests/test_ax_sweeper_plugin.py

```diff
@@ -240,3 +257,107 @@ def test_example_app(tmpdir: Path) -> None:
+        assert math.isclose(best_parameters["quadratic_args_y"], -1.0, abs_tol=1e-4)
+
+
+def test_jobs_configured_via_nested_config_with_escape_char(
+    sweep_runner: TSweepRunner,
+) -> None:
+    sweep = sweep_runner(
+        calling_file="tests/test_ax_sweeper_plugin.py",
+        calling_module=None,
+        task_function=nested_quadratic_with_escape_char,
+        config_path="config",
+        config_name="nested_quadratic_with_escape_char.yaml",
+        overrides=[
+            "hydra/sweeper=ax",
+            "hydra/launcher=basic",
+            "hydra.sweeper.params.ax_config.client.random_seed=1",
+            "hydra.sweeper.params.ax_config.max_trials=2",
+        ],
+        strict=True,
+    )
+    with sweep:
+        assert sweep.returns is None
+        returns = OmegaConf.load(f"{sweep.temp_dir}/optimization_results.yaml")
+        assert isinstance(returns, DictConfig)
+        assert returns["optimizer"] == "ax"
+        assert len(returns) == 2
+        best_parameters = returns["ax"]
+        assert len(best_parameters) == 2
+        assert math.isclose(best_parameters["quadratic_a_rgs_x"], 0.0, abs_tol=1e-4)
+        assert math.isclose(best_parameters["quadratic_a_rgs_y"], -1.0, abs_tol=1e-4)
+
+
+def test_jobs_configured_via_nested_config_with_escape_char_and_cmd(
+    sweep_runner: TSweepRunner,
+) -> None:
+    sweep = sweep_runner(
+        calling_file="tests/test_ax_sweeper_plugin.py",
+        calling_module=None,
+        task_function=nested_quadratic_with_escape_char,
+        config_path="config",
+        config_name="nested_quadratic_with_escape_char.yaml",
+        overrides=[
+            "hydra/sweeper=ax",
+            "hydra/launcher=basic",
+            "hydra.sweeper.params.ax_config.client.random_seed=1",
+            "hydra.sweeper.params.ax_config.max_trials=2",
+            "quadratic.a_rgs.x=-1:1",
+        ],
+        strict=True,
+    )
+    with sweep:
+        assert sweep.returns is None
+        returns = OmegaConf.load(f"{sweep.temp_dir}/optimization_results.yaml")
+        assert isinstance(returns, DictConfig)
+        assert returns["optimizer"] == "ax"
+        assert len(returns) == 2
+        best_parameters = returns["ax"]
+        assert len(best_parameters) == 2
+        assert math.isclose(best_parameters["quadratic_a_rgs_x"], 0.0, abs_tol=1e-4)
+        assert math.isclose(best_parameters["quadratic_a_rgs_y"], -1.0, abs_tol=1e-4)
+
+
+@pytest.mark.parametrize(
+    "inp, str_to_replace, str_to_replace_with, expected",
+    [
+        ("apple", ".", "_", "apple"),
+        ("a.pple", ".", "_", "a_pple"),
+        ("a.p.ple", ".", "_", "a_p_ple"),
+        (r"a\.pple", ".", "_", "a.pple"),
+        (r"a.p\.pl.e", ".", "_", "a_p.pl_e"),
+    ],
+)
+def test_process_key_method(
+    inp: str, str_to_replace: str, str_to_replace_with: str, expected: str
+):
+    from hydra_plugins.hydra_ax_sweeper._core import process_key
```

Comment:
just import the function at the top of the file. why import locally?

Reply:
I think it is related to https://github.com/facebookresearch/hydra/issues/476